Specify AdaptiveAvgPool-8 and AdaptiveMaxPool-8 (#5773)
openvinotoolkit/openvino: docs/ops/pooling/AdaptiveAvgPool_8.md

## AdaptiveAvgPool<a name="AdaptiveAvgPool"></a> {#openvino_docs_ops_pooling_AdaptiveAvgPool_8}

**Versioned name**: *AdaptiveAvgPool-8*

**Category**: *Pooling*

**Short description**: Applies average pooling with adaptive kernel size over the input.

> **Detailed description**: This operation calculates the output based on the 1st input and `output_size` determined by the 2nd input.

Reviewer: ```suggestion
**Detailed description**: This operation calculates the output based on the first input and `output_size` determined by the second input.
```

The kernel dimensions are calculated using the following formulae for the `NCDHW` input case:

\f[
\begin{array}{lcl}
d_{start} &=& floor(i*D_{in}/D_{out})\\
d_{end}   &=& ceil((i+1)*D_{in}/D_{out})\\
h_{start} &=& floor(j*H_{in}/H_{out})\\
h_{end}   &=& ceil((j+1)*H_{in}/H_{out})\\
w_{start} &=& floor(k*W_{in}/W_{out})\\
w_{end}   &=& ceil((k+1)*W_{in}/W_{out})
\end{array}
\f]

> The output is calculated following this formula:

Reviewer: ```suggestion
The output is calculated with the following formula:
```

\f[
Output(i,j,k) = \frac{Input[d_{start}:d_{end}, h_{start}:h_{end}, w_{start}:w_{end}]}{(d_{end}-d_{start})*(h_{end}-h_{start})*(w_{end}-w_{start})}
\f]

**Inputs**:

> *   **1**: 3D, 4D or 5D input tensor of shape `[N,C,L]`, `[N,C,H,W]` or `[N,C,D,H,W]` and type *T*. Required.

Reviewer: ```suggestion
*   **1**: 3D, 4D, or 5D input tensor of shape `[N,C,L]`, `[N,C,H,W]` or `[N,C,D,H,W]` and type *T*. Required.
```

## AdaptiveAvgPool<a name="AdaptiveAvgPool"></a> {#openvino_docs_ops_pooling_AdaptiveAvgPool_8}

**Versioned name**: *AdaptiveAvgPool-8*

**Category**: *Pooling*

**Short description**: Applies average pooling with adaptive kernel size over the input.

**Detailed description**: This operation calculates the output based on the input and `output_size`.
The kernel dimensions are calculated using the following formulae for the `NCDHW` input case:

\f[
\begin{array}{lcl}
d_{start}&=&floor(i*D_{in}/D_{out})\\
d_{end}&=&ceil((i+1)*D_{in}/D_{out})\\
h_{start}&=&floor(j*H_{in}/H_{out})\\
h_{end}&=&ceil((j+1)*H_{in}/H_{out})\\
w_{start}&=&floor(k*W_{in}/W_{out})\\
w_{end}&=&ceil((k+1)*W_{in}/W_{out})
\end{array}
\f]

The output is calculated following this formula:

\f[
Output(i,j,k)=\frac{Input[d_{start}:d_{end},h_{start}:h_{end},w_{start}:w_{end}]}{(d_{end}-d_{start})*(h_{end}-h_{start})*(w_{end}-w_{start})}
\f]

**Attributes**:

* *output_size*

  * **Description**: the target spacial output shape `[W_out]` for 3D input, `[H_out,W_out]` for 4D input,
    `[D_out,H_out,W_out]` for 5D input.

>   * **Range of values**: positive integer values

Reviewer 2: Maybe we should have not attribute `output_size`, but the input `output_size`?

Author: We should discuss this I think because `output_size` is an attribute in all frameworks which have adaptive pooling.

Reviewer 3: It seems this is an input in TensorFlow. Please double-check it.

**Versioned name**: *AdaptiveAvgPool-8*

**Category**: *Pooling*

**Short description**: Applies average pooling with adaptive kernel size over the input.

**Detailed description**: This operation calculates the output based on the 1st input and `output_size` determined by the 2nd input.
The kernel dimensions are calculated using the following formulae for the `NCDHW` input case:

\f[
\begin{array}{lcl}
d_{start} &=& floor(i*D_{in}/D_{out})\\
d_{end}   &=& ceil((i+1)*D_{in}/D_{out})\\
h_{start} &=& floor(j*H_{in}/H_{out})\\
h_{end}   &=& ceil((j+1)*H_{in}/H_{out})\\
w_{start} &=& floor(k*W_{in}/W_{out})\\
w_{end}   &=& ceil((k+1)*W_{in}/W_{out})
\end{array}
\f]

The output is calculated following this formula:

\f[
Output(i,j,k) = \frac{Input[d_{start}:d_{end}, h_{start}:h_{end}, w_{start}:w_{end}]}{(d_{end}-d_{start})*(h_{end}-h_{start})*(w_{end}-w_{start})}
\f]

**Attributes**:

* *output_type*

  * **Description**: the output tensor type
  * **Range of values**: "i64" or "i32"
  * **Type**: string
  * **Default value**: "i64"
  * **Required**: *No*

**Inputs**:

*   **1**: 3D, 4D or 5D input tensor of shape `[N,C,W]`, `[N,C,H,W]` or `[N,C,D,H,W]` and type *T*. Required.

> *   **2**: 1D tensor describing output shape for spacial dimensions. Can be `[L_out]` for 3D input, `[H_out,W_out]` for 4D input, `[D_out,H_out,W_out]` for 5D input and of type *T*. Required.

Reviewer 2: Maybe `output_shape` input should have the integer type `T_IND`, not a floating point type `T`?

## AdaptiveAvgPool<a name="AdaptiveAvgPool"></a> {#openvino_docs_ops_pooling_AdaptiveAvgPool_8}

**Versioned name**: *AdaptiveAvgPool-8*

**Category**: *Pooling*

**Short description**: Applies average pooling with adaptive kernel size over the input.

**Detailed description**: This operation calculates the output based on the 1st input and `output_size` determined by the 2nd input.
The kernel dimensions are calculated using the following formulae for the `NCDHW` input case:

\f[
\begin{array}{lcl}
d_{start} &=& floor(i*D_{in}/D_{out})\\
d_{end}   &=& ceil((i+1)*D_{in}/D_{out})\\
h_{start} &=& floor(j*H_{in}/H_{out})\\
h_{end}   &=& ceil((j+1)*H_{in}/H_{out})\\
w_{start} &=& floor(k*W_{in}/W_{out})\\
w_{end}   &=& ceil((k+1)*W_{in}/W_{out})
\end{array}
\f]

The output is calculated following this formula:

\f[
Output(i,j,k) = \frac{Input[d_{start}:d_{end}, h_{start}:h_{end}, w_{start}:w_{end}]}{(d_{end}-d_{start})*(h_{end}-h_{start})*(w_{end}-w_{start})}

> \f]

Reviewer 3: IMHO, now I am afraid that these formulas look not so obvious and can be useless for developers.
I propose:
1) I think that these formulas can be simplified by just describing pooling process for 3D tensor. For higher rank tensor it is generalized simply.
2) Avoid latex denotations that routine all stuff here because this case is simple. Also you use `/` and `\frac` for the same operations. I propose to use just `/`.

**Attributes**:

* *output_type*

  * **Description**: the output tensor type
  * **Range of values**: "i64" or "i32"
  * **Type**: string
  * **Default value**: "i64"
  * **Required**: *No*

**Inputs**:

*   **1**: 3D, 4D or 5D input tensor of shape `[N,C,W]`, `[N,C,H,W]` or `[N,C,D,H,W]` and type *T*. Required.

> *   **2**: 1D tensor describing output shape for spacial dimensions. Can be `[L_out]` for 3D input, `[H_out,W_out]` for 4D input, `[D_out,H_out,W_out]` for 5D input and of type *T_IND*. Required.

Reviewer 4: spacial -> spatial

**Outputs**:

> *   **1**: Output of type *T* and shape `[N,C,W_out]`, `[N,C,H_out,W_out]` or `[N,C,D_out,H_out,W_out]`.

Reviewer 4: Haven't you forgot to add the second output with indices which is supposed to be a user of output_type attribute?

Author: No, AvgPool doesn't have 2nd output, my mistake was adding `output_type` attribute here
